src/nscreg.Server.Common/Services/StatUnit/ElasticService.cs

```diff
@@ -317,7 +317,7 @@ public async Task<SearchVm<ElasticStatUnit>> Search(SearchQueryM filter, string
             {
                 int regionId = filter.RegionId.Value;
                 mustQueries.Add(m =>
-                    m.Term(p => p.Field(f => f.RegionId).Value(regionId))
+                    m.Term(p => p.Field(f => f.ActualAddressRegionId).Value(regionId))
```

Comment:
Search on membership in `RegionIds` instead, as that is the aggregated regions for that statunit.